src/configuration/remapper.ts

```diff
@@ -394,6 +394,19 @@ export class Remapper implements IRemapper {
           throw ForceStopRemappingError.fromVimError(vimError);
         }
 
+        if (vimState.mapDepth % 10 === 0) {
+          // Allow the user to press <C-c> or <Esc> key when inside an infinite looping remap.
+          // When inside an infinite looping recursive mapping it would block the editor until it reached
+          // the maxmapdepth. This 0ms wait allows the extension to handle any key typed by the user which
+          // means it allows the user to press <C-c> or <Esc> to force stop the looping remap.
+          // This shouldn't impact the normal use case because we're only running this every 10 nested
+          // remaps. Also when the logs are set to Error only, a looping recursive remap takes around 1.5s
+          // to reach 1000 mapDepth and give back control to the user, but when logs are set to debug it
+          // can take as long as 7 seconds.
+          const wait = (ms: number) => new Promise((res) => setTimeout(res, ms));
+          await wait(0);
+        }
```

Comment:
If we create a recursive remap does something on the buffer and then calls itself, that is working fine and allows the user to cancel it by pressing `<C-c>` or `<Esc>`. But if we create an infinite looping remap like `x -> y` and `y -> x` this blocks the extension, until it reaches the maxmapdepth (1000) and throws E223 giving control back to the user. I don't know why this happens because there is already a lot of awaits that should allow the user to press the cancel key, but the event never fires. But if I create this 0ms timeout every 10 nested remaps it then allows the user to press the cancel key. I don't know if there is a better solution to this?

Reply:
Yeah this is a bit janky but I'm not aware of a better solution, and like you said, doesn't impact normal use cases, so seems reasonable to me